xds/internal/balancer/clusterimpl/clusterimpl.go

```diff
@@ -110,7 +113,7 @@ type clusterImplBalancer struct {
 	config           *lbConfig
 	childLB          balancer.Balancer
 	cancelLoadReport func()
-	clusterName      string
+	clusterName      atomic.Value
```

Comment:
Seems an overkill to have this an atomic. How often do we expect this to change?

Reply:
Changed to mutex